astroid/nodes/node_ng.py

```diff
@@ -311,6 +310,13 @@ def frame(
 
         :returns: The first parent frame node.
         """
+        if self.parent is None:
+            if isinstance(
```

Comment:
I don't think this check should be here. If `self` is either of these `nodes` we shouldn't be in this function anyway as they have their `frame()` defined in their class definitions.

I think just raising the exception is fine.

I looked at `NamedExpr.frame()` which seems to raise `ParentMissingError`. I think if we go ahead with raising `FrameMissing` we might use this exception there as well. It shouldn't be a breaking change since `FrameMissing` inherits from `ParentMissingError`.

Reply:
The `isinstance` check is indeed unnecessary.  Just this would be enough:
```py
        if not self.parent:
            raise ParentMissingError(target=self)

        return self.parent.frame()
```

--
Technically this is a small breaking change as users could be fine with catching `AttributeError`.
We added `future=True` to `statement()` so just for completeness we should probably do that here, too. You can take a look at #1217 as example.